lib/std/io/tty.zig

```diff
@@ -7,37 +7,41 @@ const native_os = builtin.os.tag;
 
 /// Detect suitable TTY configuration options for the given file (commonly stdout/stderr).
 /// This includes feature checks for ANSI escape codes and the Windows console API, as well as
-/// respecting the `NO_COLOR` environment variable.
+/// respecting the `NO_COLOR` and `YES_COLOR` environment variables to override the default.
 pub fn detectConfig(file: File) Config {
-    return detectConfigForce(file, false);
+    const force_color: ?bool = if (builtin.os.tag == .wasi)
+        null // wasi does not support environment variables
+    else if (process.hasEnvVarConstant("NO_COLOR"))
+        false
+    else if (process.hasEnvVarConstant("YES_COLOR"))
+        true
+    else
+        null;
+    return detectConfigForce(file, force_color);
 }
 
-/// Detect suitable TTY configuration options, but force the use of color if the platform
-/// supports it. On platforms such as wasi that do not support ANSI codes, if they are
-/// desired for some reason just use `Config.escape_codes` directly.
-pub fn detectConfigForce(file: File, force_color: bool) Config {
-    if (builtin.os.tag == .wasi) {
-        // Per https://github.com/WebAssembly/WASI/issues/162 ANSI codes
-        // aren't currently supported.
-        return .no_color;
```

Comment:
I originally had this behaviour (forcing colour on WASI returning `.escape_codes` but I think the least surprising behaviour is for `detectConfigForce` to not output escapes codes on platforms that don't support them. If someone actually wants them in an environment that doesn't support them they should just _not_ call `detectConfig*`.

Reply:
When color is forced on, it is irrelevant whether a platform supports it or not. The intention of forcing it on is that the output stream will be interpreted to have escape codes in them, and handled appropriately. This can be done on WASI, or Windows, or any platform.